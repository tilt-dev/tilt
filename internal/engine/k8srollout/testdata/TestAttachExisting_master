
Attaching to existing pod (pod-id). Only new logs will be streamed.
