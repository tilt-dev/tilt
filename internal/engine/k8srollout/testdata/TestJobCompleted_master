
Tracking new pod rollout (pod-id):
     ┊ Scheduled       - 1s
     ┊ Initialized     - 4s
     ┊ Completed       - 5s
