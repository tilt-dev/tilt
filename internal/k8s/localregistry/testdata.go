package localregistry

// Copied verbatim from
// https://github.com/kubernetes/enhancements/blob/0d69f7cea6fbe73a7d70fab569c6898f5ccb7be0/keps/sig-cluster-lifecycle/generic/1755-communicating-a-local-registry/README.md

const SampleConfigMap = `
apiVersion: v1
kind: ConfigMap
metadata:
  name: local-registry-hosting
  namespace: kube-public
data:
  localRegistryHosting.v1: |
    host: "localhost:5000"
    hostFromContainerRuntime: "registry:5000"
    hostFromClusterNetwork: "kind-registry:5000"
    help: "https://kind.sigs.k8s.io/docs/user/local-registry/"
`
