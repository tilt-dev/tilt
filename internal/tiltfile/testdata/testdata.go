package testdata

import (
	"path/filepath"
	"runtime"
)

func NginxIngressChartPath() string {
	return filepath.Join(staticPath(), "nginx-ingress-0.31.0.tgz")
}

func staticPath() string {
	_, file, _, ok := runtime.Caller(0)
	if !ok {
		panic("Could not locate path to tiltfile/testdata")
	}

	return filepath.Dir(file)
}
