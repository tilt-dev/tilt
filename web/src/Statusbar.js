import React, { Component } from 'react';
import './Statusbar.scss';

class Statusbar extends Component {
  render() {
    return (<div className="Statusbar">I'm a statusbar!</div>)
  }
}

export default Statusbar;
