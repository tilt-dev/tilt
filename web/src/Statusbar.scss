@import "constants";

.Statusbar {
  position: fixed;
  left: 0;
  right: 0;
  bottom: 0;
  z-index: $z-statusbar;
  height: $statusbar-height;
  background-color: $color-white;
  color: $color-gray;
  display: flex;
  align-items: stretch;
  border-top: 1px solid $color-gray-darkest;
  border-bottom: 1px solid $color-gray-darkest;
  box-sizing: border-box;
}

.Statusbar-panel {
  display: flex;
  align-items: center;
  box-sizing: border-box;
}
.Statusbar-panel + .Statusbar-panel {
  border-left: 1px dashed $color-gray-lightest;
}

// Errors and Warnings
.Statusbar-errWarnPanel {
  display: flex;
  align-items: stretch;
}
.Statusbar-errWarnPanel-child {
  display: flex;
  align-items: center;
  padding-left: $spacing-unit / 2;
  padding-right: $spacing-unit / 2;
  // Eyeball to make them the same width, and not change size when contents update
  min-width: $spacing-unit * 3.5;
}
.Statusbar-errWarnPanel-child + .Statusbar-errWarnPanel-child {
  border-left: 1px solid rgba($color-gray-lightest, $translucent);
}
.Statusbar-errWarnPanel-icon {
  fill: $color-gray-lightest;
  opacity: $translucent;
  margin-right: $spacing-unit / 3;
  filter: drop-shadow(-1px 1px 0px rgba($color-gray-darkest, $translucent-ish));
}
.Statusbar-errWarnPanel-icon--error {
  fill: $color-red;
  opacity: 1;
  filter: none;
}
.Statusbar-errWarnPanel-icon--warning {
  fill: $color-yellow;
  opacity: 1;
  filter: none;
}
.Statusbar-errWarnPanel-count {
  font-weight: bold;
}

// Status Message
.Statusbar-statusMsgPanel {
  flex-grow: 1;
  display: flex;
  justify-content: space-between;
  padding-left: $spacing-unit / 2;
  padding-right: $spacing-unit / 2;
}

.Statusbar-statusMsgPanel-label {
  text-transform: uppercase;
  font-size: $font-size-small;
  font-weight: bold;
  color: $color-gray-lightest;
  margin-right: $spacing-unit / 4;
}

// Progress
.Statusbar-progressPanel {
  width: $sidebar-width - $statusbar-tiltPanel-width;
  max-width: $sidebar-maxWidth;
  justify-content: space-between;
  display: flex;
  padding-left: $spacing-unit / 2;
  box-sizing: border-box;
}

// Tilt Logo and Update
.Statusbar-tiltPanel {
  display: flex;
  align-items: center;
  padding-left: $spacing-unit / 2;
  padding-right: $spacing-unit / 2;
  height: $statusbar-height;
  width: $statusbar-tiltPanel-width;
  box-sizing: border-box;
}

.Statusbar-tiltPanel-updateIcon {
  display: flex;
  position: absolute;
  outline: $color-white;
  transition: transform $animation-timing ease;
  margin-left: -6px;
  margin-top: -6px;
}

.Statusbar-tiltPanel-link {
  display: flex;
}

.Statusbar-tiltPanel-upgradeTooltip {
  position: absolute;
  right: 0;
  width: $sidebar-width;
  height: 2 * $statusbar-height;
  top: -2 * $statusbar-height;
  background: $color-gray-dark;
  color: $color-white;
  border: 2px solid $color-white;
  padding: $spacing-unit / 3;
  box-sizing: border-box;
  display: flex;
  align-items: center;
  justify-content: center;
  opacity: 0;
  pointer-events: none;
  transition: opacity $animation-timing ease;
}

.Statusbar-tiltPanel-link:hover .Statusbar-tiltPanel-upgradeTooltip {
  opacity: 1;
}

.Statusbar-tiltPanel-link:hover .Statusbar-tiltPanel-updateIcon {
  transform: rotate(180deg);
}