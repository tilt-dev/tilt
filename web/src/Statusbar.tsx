import React, { PureComponent, ReactElement } from "react"
import { ReactComponent as LogoSvg } from "./assets/svg/logo.svg"
import { ReactComponent as ErrorSvg } from "./assets/svg/error.svg"
import { ReactComponent as WarningSvg } from "./assets/svg/warning.svg"
import { ReactComponent as UpdateAvailableSvg } from "./assets/svg/update-available.svg"
import { combinedStatus, warnings } from "./status"
import "./Statusbar.scss"
import { combinedStatusMessage } from "./combinedStatusMessage"
import { ResourceStatus } from "./types"
import mostRecentBuildToDisplay from "./mostRecentBuild"
import { Link } from "react-router-dom"

type Build = Proto.webviewBuildRecord
type TiltBuild = Proto.webviewTiltBuild

class StatusItem {
  public warningCount: number = 0
  public up: boolean = false
  public hasError: boolean = false
  public name: string
  public currentBuild: Build
  public lastBuild: Build | null
  public podStatus: string
  public podStatusMessage: string
  public pendingBuildSince: string
  public buildHistory: Array<Build>
  public pendingBuildEdits: Array<string>

  /**
   * Create a pared down StatusItem from a ResourceView
   */
  constructor(res: any) {
    this.name = res.name
    this.warningCount = warnings(res).length

    let status = combinedStatus(res)
    this.up = status === ResourceStatus.Healthy
    this.hasError = status === ResourceStatus.Unhealthy
    this.currentBuild = res.currentBuild
    this.buildHistory = res.buildHistory
    this.lastBuild = res.buildHistory ? res.buildHistory[0] : null
    this.podStatus = res.k8sResourceInfo && res.k8sResourceInfo.podStatus
    this.podStatusMessage =
      res.k8sResourceInfo && res.k8sResourceInfo.podStatusMessage
    this.pendingBuildSince = res.pendingBuildSince
    this.pendingBuildEdits = res.pendingBuildEdits
  }
}

type StatusBarProps = {
  items: Array<StatusItem>
  alertsUrl: string
  runningVersion: TiltBuild | null | undefined
  latestVersion: TiltBuild | null | undefined
  checkVersion: boolean
}

class Statusbar extends PureComponent<StatusBarProps> {
  errorWarningPanel(errorCount: number, warningCount: number) {
    return (
      <section className="Statusbar-panel Statusbar-errWarnPanel">
        <div className="Statusbar-errWarnPanel-child">
          <ErrorSvg
            className={`Statusbar-errWarnPanel-icon ${
              errorCount > 0 ? "Statusbar-errWarnPanel-icon--error" : ""
            }`}
          />
          <p>
            <span className="Statusbar-errWarnPanel-count Statusbar-errWarnPanel-count--error">
              {errorCount}
            </span>{" "}
            <Link to={this.props.alertsUrl}>
              error
              {errorCount === 1 ? "" : "s"}
            </Link>
          </p>
        </div>
        <div className="Statusbar-errWarnPanel-child">
          <WarningSvg
            className={`Statusbar-errWarnPanel-icon ${
              warningCount > 0 ? "Statusbar-errWarnPanel-icon--warning" : ""
            }`}
          />
          <p>
            <span className="Statusbar-errWarnPanel-count">{warningCount}</span>{" "}
            warning
            {warningCount === 1 ? "" : "s"}
          </p>
        </div>
      </section>
    )
  }

  progressPanel(upCount: number, itemCount: number) {
    return (
      <section className="Statusbar-panel Statusbar-progressPanel">
        <p>
          <strong>{upCount}</strong>/{itemCount} running
        </p>
      </section>
    )
  }

  statusMessagePanel(build: any, edits: string) {
    let lastEdit = <span>—</span>
    if (build && edits) {
      lastEdit = (
        <span>
          <span className="LastEditStatus-name" data-tip={edits}>
            {build.name}
          </span>
          <span className="LastEditStatus-details">{" ‣ " + edits}</span>
        </span>
      )
    }
    return (
      <section className="Statusbar-panel Statusbar-statusMsgPanel">
        <div>{combinedStatusMessage(this.props.items)}</div>
        <div className="LastEditStatus">
          <span className="LastEditStatus-label">Last Edit:</span>
          {lastEdit}
        </div>
      </section>
    )
  }

  tiltPanel(
    runningVersion: TiltBuild | null | undefined,
    latestVersion: TiltBuild | null | undefined,
    shouldCheckVersion: boolean
  ) {
    let content: ReactElement = <LogoSvg className="Statusbar-logo" />
    if (
      shouldCheckVersion &&
      latestVersion &&
      latestVersion.version &&
      runningVersion &&
      runningVersion.version &&
      !runningVersion.dev &&
      runningVersion.version !== latestVersion.version
    ) {
      content = (
        <a
          href="https://docs.tilt.dev/upgrade.html"
          className="Statusbar-tiltPanel-link"
          target="_blank"
          rel="noopener noreferrer"
        >
          <p className="Statusbar-tiltPanel-upgradeTooltip">
            <span role="img" aria-label="Decorative sparkling stars">
              ✨
            </span>
            Get Tilt v{latestVersion.version}!{" "}
            <span role="img" aria-label="Decorative sparkling stars">
              ✨
            </span>
            <br />
            (You're running v{runningVersion.version})
          </p>
          {content}
          <UpdateAvailableSvg />
        </a>
      )
    }

    return <section className="Statusbar-tiltPanel">{content}</section>
  }

  render() {
    let errorCount = 0
    let warningCount = 0
    let upCount = 0

    let items = this.props.items
    items.forEach(item => {
      if (item.hasError) {
        errorCount++
      }
      if (item.up) {
        upCount++
      }
      warningCount += item.warningCount
    })
    let errorWarningPanel = this.errorWarningPanel(errorCount, warningCount)

    let build = mostRecentBuildToDisplay(this.props.items)
    let editMessage = ""
    if (build && build.edits.length > 0) {
      editMessage = `${build.edits[0]}`
      if (build.edits.length > 1) {
        editMessage += `[+ ${build.edits.length - 1} more]`
      }
    }
    let statusMessagePanel = this.statusMessagePanel(build, editMessage)

    let resCount = items.length
    let progressPanel = this.progressPanel(upCount, resCount)

    let tiltPanel = this.tiltPanel(
      this.props.runningVersion,
      this.props.latestVersion,
      this.props.checkVersion
    )

    return (
      <div className="Statusbar">
        {errorWarningPanel}
        {statusMessagePanel}
        {progressPanel}
        {tiltPanel}
      </div>
    )
  }
}

export default Statusbar

export { StatusItem }
