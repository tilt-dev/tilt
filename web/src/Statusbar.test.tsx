import React from "react"
import renderer from "react-test-renderer"
import Statusbar, { StatusItem } from "./Statusbar"
import { mount } from "enzyme"
import { twoResourceView } from "./testdata"
import { MemoryRouter } from "react-router"

type TiltBuild = Proto.webviewTiltBuild

describe("StatusBar", () => {
  let runningVersion: TiltBuild = {
    version: "v0.8.1",
    date: "1970-01-01",
    dev: false,
  }
  it("renders without crashing", () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <Statusbar
            items={[]}
            alertsUrl="/alerts"
            runningVersion={runningVersion}
            latestVersion={null}
            checkVersion={true}
          />
        </MemoryRouter>
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  it("renders two items both errors", () => {
    let items = twoResourceView().resources.map((res: any) => {
      res.currentBuild = {}
      res.pendingBuildSince = ""
      return new StatusItem(res)
    })
    let statusbar = mount(
      <MemoryRouter>
        <Statusbar
          items={items}
          alertsUrl="/alerts"
          runningVersion={runningVersion}
          latestVersion={null}
          checkVersion={true}
        />
      </MemoryRouter>
    )
    expect(
      statusbar.find(".Statusbar-errWarnPanel-count--error").html()
    ).toContain("2")

    statusbar.unmount()
  })

  it("renders two items both errors snapshot", () => {
    let items = twoResourceView().resources.map((res: any) => {
      res.currentBuild = {}
      res.pendingBuildSince = ""
      return new StatusItem(res)
    })
    const tree = renderer
      .create(
        <MemoryRouter>
          <Statusbar
            items={items}
            alertsUrl="/alerts"
            runningVersion={runningVersion}
            latestVersion={null}
            checkVersion={true}
          />
        </MemoryRouter>
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  it("renders two items all ok snapshot", () => {
    let view = twoResourceView()
    view.resources.forEach((res: any) => {
      res.buildHistory[0].error = ""
    })

    let items = view.resources.map((res: any) => new StatusItem(res))
    const tree = renderer
      .create(
        <MemoryRouter>
          <Statusbar
            items={items}
            alertsUrl="/alerts"
            runningVersion={runningVersion}
            latestVersion={null}
            checkVersion={true}
          />
        </MemoryRouter>
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  it("renders two items all ok", () => {
    let view = twoResourceView()
    view.resources.forEach((res: any) => {
      res.buildHistory[0].error = ""
    })
    let items = view.resources.map((res: any) => new StatusItem(res))
    let statusbar = mount(
      <MemoryRouter>
        <Statusbar
          items={items}
          alertsUrl="/alerts"
          runningVersion={runningVersion}
          latestVersion={null}
          checkVersion={true}
        />
      </MemoryRouter>
    )
    expect(
      statusbar.find(".Statusbar-errWarnPanel-count--error").html()
    ).toContain("0")

    statusbar.unmount()
  })

  it("renders an upgrade badge when the version is out of date", () => {
    let view = twoResourceView()
    view.resources.forEach((res: any) => {
      res.buildHistory[0].error = ""
    })
    let items = view.resources.map((res: any) => new StatusItem(res))
    let latestVersion = runningVersion
    latestVersion.version = "10.0.0"
    const tree = renderer
      .create(
        <MemoryRouter>
          <Statusbar
            items={items}
            alertsUrl="/alerts"
            runningVersion={runningVersion}
            latestVersion={latestVersion}
            checkVersion={true}
          />
        </MemoryRouter>
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  it("does not render an upgrade badge when there is no latestVersion", () => {
    let view = twoResourceView()
    view.resources.forEach((res: any) => {
      res.buildHistory[0].error = ""
    })
    let items = view.resources.map((res: any) => new StatusItem(res))
    let latestVersion = { version: "", date: "", dev: false }
    const tree = renderer
      .create(
        <MemoryRouter>
          <Statusbar
            items={items}
            alertsUrl="/alerts"
            runningVersion={runningVersion}
            latestVersion={latestVersion}
            checkVersion={true}
          />
        </MemoryRouter>
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  it("does not render an upgrade badge when runningVersion is dev", () => {
    let view = twoResourceView()
    view.resources.forEach((res: any) => {
      res.buildHistory[0].error = ""
    })
    let items = view.resources.map((res: any) => new StatusItem(res))
    let latestVersion = runningVersion
    latestVersion.version = "10.0.0"
    let devRunningVersion = runningVersion
    devRunningVersion.dev = true
    const tree = renderer
      .create(
        <MemoryRouter>
          <Statusbar
            items={items}
            alertsUrl="/alerts"
            runningVersion={devRunningVersion}
            latestVersion={latestVersion}
            checkVersion={true}
          />
        </MemoryRouter>
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  it("does not render an upgrade badge when the version is out of date if version check is disabled", () => {
    let view = twoResourceView()
    view.resources.forEach((res: any) => {
      res.buildHistory[0].error = ""
    })
    let items = view.resources.map((res: any) => new StatusItem(res))
    let latestVersion = runningVersion
    latestVersion.version = "10.0.0"
    const root = mount(
      <MemoryRouter>
        <Statusbar
          items={items}
          alertsUrl="/alerts"
          runningVersion={runningVersion}
          latestVersion={latestVersion}
          checkVersion={false}
        />
      </MemoryRouter>
    )

    let el = root.find(".Statusbar-tiltPanel-upgradeTooltip")
    expect(el).toHaveLength(0)
  })
})

describe("StatusItem", () => {
  it("can be constructed with no build history", () => {
    let si = new StatusItem({})
    expect(si.hasError).toBe(false)
  })
})
