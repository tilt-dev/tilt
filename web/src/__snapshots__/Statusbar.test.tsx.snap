// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatusBar does not render an upgrade badge when runningVersion is dev 1`] = `
<div
  className="Statusbar"
>
  <section
    className="Statusbar-panel Statusbar-errWarnPanel"
  >
    <div
      className="Statusbar-errWarnPanel-child"
    >
      <svg
        className="Statusbar-errWarnPanel-icon "
      >
        error.svg
      </svg>
      <p>
        <span
          className="Statusbar-errWarnPanel-count Statusbar-errWarnPanel-count--error"
        >
          0
        </span>
         
        <a
          href="/alerts"
          onClick={[Function]}
        >
          error
          s
        </a>
      </p>
    </div>
    <div
      className="Statusbar-errWarnPanel-child"
    >
      <svg
        className="Statusbar-errWarnPanel-icon "
      >
        warning.svg
      </svg>
      <p>
        <span
          className="Statusbar-errWarnPanel-count"
        >
          0
        </span>
         
        warning
        s
      </p>
    </div>
  </section>
  <section
    className="Statusbar-panel Statusbar-statusMsgPanel"
  >
    <div>
      Updating vigoda…
    </div>
    <div
      className="LastEditStatus"
    >
      <span
        className="LastEditStatus-label"
      >
        Last Edit:
      </span>
      <span>
        <span
          className="LastEditStatus-name"
          data-tip="main.go[+ 2 more]"
        >
          vigoda
        </span>
        <span
          className="LastEditStatus-details"
        >
           ‣ main.go[+ 2 more]
        </span>
      </span>
    </div>
  </section>
  <section
    className="Statusbar-panel Statusbar-progressPanel"
  >
    <p>
      <strong>
        0
      </strong>
      /
      2
       running
    </p>
  </section>
  <section
    className="Statusbar-tiltPanel"
  >
    <svg
      className="Statusbar-logo"
    >
      logo.svg
    </svg>
  </section>
</div>
`;

exports[`StatusBar does not render an upgrade badge when there is no latestVersion 1`] = `
<div
  className="Statusbar"
>
  <section
    className="Statusbar-panel Statusbar-errWarnPanel"
  >
    <div
      className="Statusbar-errWarnPanel-child"
    >
      <svg
        className="Statusbar-errWarnPanel-icon "
      >
        error.svg
      </svg>
      <p>
        <span
          className="Statusbar-errWarnPanel-count Statusbar-errWarnPanel-count--error"
        >
          0
        </span>
         
        <a
          href="/alerts"
          onClick={[Function]}
        >
          error
          s
        </a>
      </p>
    </div>
    <div
      className="Statusbar-errWarnPanel-child"
    >
      <svg
        className="Statusbar-errWarnPanel-icon "
      >
        warning.svg
      </svg>
      <p>
        <span
          className="Statusbar-errWarnPanel-count"
        >
          0
        </span>
         
        warning
        s
      </p>
    </div>
  </section>
  <section
    className="Statusbar-panel Statusbar-statusMsgPanel"
  >
    <div>
      Updating vigoda…
    </div>
    <div
      className="LastEditStatus"
    >
      <span
        className="LastEditStatus-label"
      >
        Last Edit:
      </span>
      <span>
        <span
          className="LastEditStatus-name"
          data-tip="main.go[+ 2 more]"
        >
          vigoda
        </span>
        <span
          className="LastEditStatus-details"
        >
           ‣ main.go[+ 2 more]
        </span>
      </span>
    </div>
  </section>
  <section
    className="Statusbar-panel Statusbar-progressPanel"
  >
    <p>
      <strong>
        0
      </strong>
      /
      2
       running
    </p>
  </section>
  <section
    className="Statusbar-tiltPanel"
  >
    <svg
      className="Statusbar-logo"
    >
      logo.svg
    </svg>
  </section>
</div>
`;

exports[`StatusBar renders an upgrade badge when the version is out of date 1`] = `
<div
  className="Statusbar"
>
  <section
    className="Statusbar-panel Statusbar-errWarnPanel"
  >
    <div
      className="Statusbar-errWarnPanel-child"
    >
      <svg
        className="Statusbar-errWarnPanel-icon "
      >
        error.svg
      </svg>
      <p>
        <span
          className="Statusbar-errWarnPanel-count Statusbar-errWarnPanel-count--error"
        >
          0
        </span>
         
        <a
          href="/alerts"
          onClick={[Function]}
        >
          error
          s
        </a>
      </p>
    </div>
    <div
      className="Statusbar-errWarnPanel-child"
    >
      <svg
        className="Statusbar-errWarnPanel-icon "
      >
        warning.svg
      </svg>
      <p>
        <span
          className="Statusbar-errWarnPanel-count"
        >
          0
        </span>
         
        warning
        s
      </p>
    </div>
  </section>
  <section
    className="Statusbar-panel Statusbar-statusMsgPanel"
  >
    <div>
      Updating vigoda…
    </div>
    <div
      className="LastEditStatus"
    >
      <span
        className="LastEditStatus-label"
      >
        Last Edit:
      </span>
      <span>
        <span
          className="LastEditStatus-name"
          data-tip="main.go[+ 2 more]"
        >
          vigoda
        </span>
        <span
          className="LastEditStatus-details"
        >
           ‣ main.go[+ 2 more]
        </span>
      </span>
    </div>
  </section>
  <section
    className="Statusbar-panel Statusbar-progressPanel"
  >
    <p>
      <strong>
        0
      </strong>
      /
      2
       running
    </p>
  </section>
  <section
    className="Statusbar-tiltPanel"
  >
    <svg
      className="Statusbar-logo"
    >
      logo.svg
    </svg>
  </section>
</div>
`;

exports[`StatusBar renders two items all ok snapshot 1`] = `
<div
  className="Statusbar"
>
  <section
    className="Statusbar-panel Statusbar-errWarnPanel"
  >
    <div
      className="Statusbar-errWarnPanel-child"
    >
      <svg
        className="Statusbar-errWarnPanel-icon "
      >
        error.svg
      </svg>
      <p>
        <span
          className="Statusbar-errWarnPanel-count Statusbar-errWarnPanel-count--error"
        >
          0
        </span>
         
        <a
          href="/alerts"
          onClick={[Function]}
        >
          error
          s
        </a>
      </p>
    </div>
    <div
      className="Statusbar-errWarnPanel-child"
    >
      <svg
        className="Statusbar-errWarnPanel-icon "
      >
        warning.svg
      </svg>
      <p>
        <span
          className="Statusbar-errWarnPanel-count"
        >
          0
        </span>
         
        warning
        s
      </p>
    </div>
  </section>
  <section
    className="Statusbar-panel Statusbar-statusMsgPanel"
  >
    <div>
      Updating vigoda…
    </div>
    <div
      className="LastEditStatus"
    >
      <span
        className="LastEditStatus-label"
      >
        Last Edit:
      </span>
      <span>
        <span
          className="LastEditStatus-name"
          data-tip="main.go[+ 2 more]"
        >
          vigoda
        </span>
        <span
          className="LastEditStatus-details"
        >
           ‣ main.go[+ 2 more]
        </span>
      </span>
    </div>
  </section>
  <section
    className="Statusbar-panel Statusbar-progressPanel"
  >
    <p>
      <strong>
        0
      </strong>
      /
      2
       running
    </p>
  </section>
  <section
    className="Statusbar-tiltPanel"
  >
    <svg
      className="Statusbar-logo"
    >
      logo.svg
    </svg>
  </section>
</div>
`;

exports[`StatusBar renders two items both errors snapshot 1`] = `
<div
  className="Statusbar"
>
  <section
    className="Statusbar-panel Statusbar-errWarnPanel"
  >
    <div
      className="Statusbar-errWarnPanel-child"
    >
      <svg
        className="Statusbar-errWarnPanel-icon Statusbar-errWarnPanel-icon--error"
      >
        error.svg
      </svg>
      <p>
        <span
          className="Statusbar-errWarnPanel-count Statusbar-errWarnPanel-count--error"
        >
          2
        </span>
         
        <a
          href="/alerts"
          onClick={[Function]}
        >
          error
          s
        </a>
      </p>
    </div>
    <div
      className="Statusbar-errWarnPanel-child"
    >
      <svg
        className="Statusbar-errWarnPanel-icon "
      >
        warning.svg
      </svg>
      <p>
        <span
          className="Statusbar-errWarnPanel-count"
        >
          0
        </span>
         
        warning
        s
      </p>
    </div>
  </section>
  <section
    className="Statusbar-panel Statusbar-statusMsgPanel"
  >
    <div>
      Build failed: vigoda
    </div>
    <div
      className="LastEditStatus"
    >
      <span
        className="LastEditStatus-label"
      >
        Last Edit:
      </span>
      <span>
        <span
          className="LastEditStatus-name"
          data-tip="main.go[+ 1 more]"
        >
          snack
        </span>
        <span
          className="LastEditStatus-details"
        >
           ‣ main.go[+ 1 more]
        </span>
      </span>
    </div>
  </section>
  <section
    className="Statusbar-panel Statusbar-progressPanel"
  >
    <p>
      <strong>
        0
      </strong>
      /
      2
       running
    </p>
  </section>
  <section
    className="Statusbar-tiltPanel"
  >
    <svg
      className="Statusbar-logo"
    >
      logo.svg
    </svg>
  </section>
</div>
`;

exports[`StatusBar renders without crashing 1`] = `
<div
  className="Statusbar"
>
  <section
    className="Statusbar-panel Statusbar-errWarnPanel"
  >
    <div
      className="Statusbar-errWarnPanel-child"
    >
      <svg
        className="Statusbar-errWarnPanel-icon "
      >
        error.svg
      </svg>
      <p>
        <span
          className="Statusbar-errWarnPanel-count Statusbar-errWarnPanel-count--error"
        >
          0
        </span>
         
        <a
          href="/alerts"
          onClick={[Function]}
        >
          error
          s
        </a>
      </p>
    </div>
    <div
      className="Statusbar-errWarnPanel-child"
    >
      <svg
        className="Statusbar-errWarnPanel-icon "
      >
        warning.svg
      </svg>
      <p>
        <span
          className="Statusbar-errWarnPanel-count"
        >
          0
        </span>
         
        warning
        s
      </p>
    </div>
  </section>
  <section
    className="Statusbar-panel Statusbar-statusMsgPanel"
  >
    <div>
      
    </div>
    <div
      className="LastEditStatus"
    >
      <span
        className="LastEditStatus-label"
      >
        Last Edit:
      </span>
      <span>
        —
      </span>
    </div>
  </section>
  <section
    className="Statusbar-panel Statusbar-progressPanel"
  >
    <p>
      <strong>
        0
      </strong>
      /
      0
       running
    </p>
  </section>
  <section
    className="Statusbar-tiltPanel"
  >
    <svg
      className="Statusbar-logo"
    >
      logo.svg
    </svg>
  </section>
</div>
`;
